src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+            eState += "ValvePneumaticSuspension error: " + vpsEState
+        if manEState not in ("0", 0) + EMPTY_VALUES:
+            if not eState == "":
+                eState += ", "
+            eState += "Manometer2 error: " + manEState
+        valve_state = int(self._valve.Actual)
+        if valve_state == VALVE_ERROR:  # in case of valve error
+            if not eState == "":
+                eState += ", "
+            eState += "ValvePneumaticSuspension is in error"
+        elif valve_state == VALVE_UNDEF:  # in case no communication is present with the valve
+            if not eState == "":
+                eState += ", "
+            eState += "ValvePneumaticSuspension could not be contacted"
+        if eState == "":
+            self.state._set_value(model.ST_RUNNING, force_write=True)
+        else:
+            self.state._set_value(HwError(eState), force_write=True)
+
+    def _changeValve(self, goal):
+        """
+        goal (bool): goal position of the valve: (True: "open", False: "closed")
+        return (bool): goal position of the valve set to the server: (True: "open", False: "closed")
+
+        Opens or closes the valve.
+        Returns True if the valve is opened, False otherwise
+        """
+        logging.debug("Setting valve to %s." % goal)
+        self._valve.Target = VALVE_OPEN if goal else VALVE_CLOSED
+        return self._valve.Target == VALVE_OPEN
+
+    def terminate(self):
+        """
+        Called when Odemis is closed
+        """
+        self.parent.datamodel.HybridPlatform.ValvePneumaticSuspension.ErrorState.Unsubscribe(self._updateErrorState)
+        self.parent.datamodel.HybridPlatform.Manometer2.ErrorState.Unsubscribe(self._updateErrorState)
+        self._valve.Unsubscribe(self._updatePower)
+        self._gauge.Unsubscribe(self._updatePressure)
+        self._valve = None
+        self._gauge = None
+
+
+class vacuumChamber(model.Actuator):
+    """
+    This represents the vacuum chamber from Orsay Physics
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Has axes:
+        • "vacuum": unit is "None", choices is {0 : "vented", 1 : "primary vacuum", 2 : "high vacuum"}
+
+        Defines the following VA's and links them to the callbacks from the Orsay server:
+        • gateOpen (BooleanVA, set to True to open/start and False to close/stop)
+        • position (VA, read-only, value is {"vacuum" : _chamber.VacuumStatus.Actual})
+        • pressure (FloatContinuous, range=(0, 110000), read-only, unit is "Pa", value is _chamber.Pressure.Actual)
+        """
+
+        axes = {"vacuum": model.Axis(unit=None, choices={0: "vented", 1: "primary vacuum", 2: "high vacuum"})}
+
+        model.Actuator.__init__(self, name, role, parent=parent, axes=axes, **kwargs)
+
+        self._gate = None
+        self._chamber = None
+
+        self.pressure = model.FloatContinuous(0.0, range=(0, 110000), readonly=True, unit="Pa")
```

Comment:
Doest the parameter provide a .Min and .Max? If so, please use it in the range. Otherwise, please move the range to a constant at the top of the file, like:
`VACUUM_CHAMBER_PRESSURE_RNG = (0, 110000)  # Pa`